src/pages/therapists/index.tsx

```diff
@@ -1,34 +1,140 @@
-import { Container, InputBase, IconButton } from '@mui/material';
-import Box from '@mui/material/Box';
+import { useState, useEffect, ChangeEvent } from 'react';
+import {
+  Container, InputBase, IconButton, Alert, Skeleton, Box, Grid, Pagination,
+} from '@mui/material';
 import SearchIcon from '@mui/icons-material/Search';
-import Grid from '@mui/material/Grid';
-import GridCard from '../../components/TherapistCard';
+import axios from 'axios';
+import { TherapistCardType, GridCard } from '../../components';
+import { SearchBoxStyle, SkeletonBoxStyle } from './classes';
 
-const Therapist = () => (
-  <>
-    <Container>Therapist Page</Container>
-    <Box sx={{ flexGrow: 1 }}>
+const Therapist = () => {
+  const [therapists, setTherapists] = useState([]);
+  const [searchQuery, setSearchQuery] = useState('');
+  const [currentPage, setCurrentPage] = useState(1);
+  const [totalPages, setTotalPages] = useState(1);
+  const [loading, setLoading] = useState(true);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        setLoading(true);
+        const response = await axios.get('/api/v1/therapists', {
+          params: {
+            q: searchQuery,
+            page: currentPage,
+          },
+        });
+        const { data } = response;
+        setTherapists(data.data.rows);
+        setTotalPages(data.pagination.totalPages);
+        setLoading(false);
+      } catch (error) {
+        // eslint-disable-next-line no-console
+        console.error('Error fetching therapists:', error);
```

Comment:
the case that we might receive errors is a server or validation error, So instead of consol.error why not use `enqueSnackbar` to alert unexpected error message